lib/generator.js

```diff
@@ -1034,37 +1042,46 @@
         executionDeferred.promise.fail(function (err) {
             jsDeferred.reject(err);
             pixmapDeferred.reject(err);
+            profileDeferred.reject(err);
         });
 
         // Resolve the pixmapDeferred now if we aren't actually expecting a pixmap
         if (params.boundsOnly) {
             pixmapDeferred.resolve();
+            profileDeferred.resolve();
+        }
+        
+        // Resolve the profileDefferred if we aren't expecting it to come back 
+        if (!params.getICCProfileData) {
+            profileDeferred.resolve();
         }
 
-        Q.all([jsDeferred.promise, pixmapDeferred.promise]).spread(
-            function (js, pixmapBuffer) {
+        return Q.all([jsDeferred.promise, profileDeferred.promise, pixmapDeferred.promise]).spread(
+            function (js, iccProfileBuffer, pixmapBuffer) {
                 executionDeferred.resolve();
 
                 if (params.boundsOnly && js && js.bounds) {
-                    overallDeferred.resolve(js);
+                    return js;
                 } else if (js && js.bounds && pixmapBuffer) {
                     var pixmap = xpm.Pixmap(pixmapBuffer);
                     pixmap.bounds = js.bounds;
-                    overallDeferred.resolve(pixmap);
+                    if (iccProfileBuffer) {
+                        pixmap.iccProfile = iccProfileBuffer;
+                    }
+                    return pixmap;
                 } else {
                     var errStr = "Unexpected response from PS in getLayerPixmap: jsDeferred val: " +
                         JSON.stringify(js) +
+                        ", iccProfileBuffer was expected: " + params.getICCProfileData + ", val: " +
+                        iccProfileBuffer ? "truthy" : "falsy" +
                         ", pixmapDeferred val: " +
                         pixmapBuffer ? "truthy" : "falsy";
-                    overallDeferred.reject(new Error(errStr));
+                    throw new Error(errStr);
                 }
             }, function (err) {
                 executionDeferred.reject(err);
-                overallDeferred.reject(err);
```

Comment:
It seems like a *direct* refactoring would mean that you should throw here as well ... but maybe you determined that wasn't necessary?